### PR TITLE
Prepare patch release v10.0.1

### PR DESCRIPTION
**Pull request by kiryldz**
PRs must be submitted under the terms of our Contributor License Agreement [CLA](https://github.com/mapbox/mapbox-maps-android/blob/main/CONTRIBUTING.md#contributor-license-agreement).
Fixes: < Link to related issues that will be fixed by this pull request, if they exist >

## Pull request checklist:
 - [ ] Briefly describe the changes in this PR.
 - [ ] Include before/after visuals or gifs if this PR includes visual changes.
    
 - [ ] Write tests for all new functionality. If tests were not written, please explain why.
 - [ ] Optimize code for java consumption (`@JvmOverloads`, `@file:JvmName`, etc).
 - [ ] Add example if relevant.
 - [ ] Document any changes to public APIs.
 - [ ] Apply changelog label ('breaking change', 'bug :beetle:', 'build', 'docs', 'feature :green_apple:', 'performance :zap:', 'testing :100:') or use the label 'skip changelog'
 - [ ] Add an entry inside this element for inclusion in the `mapbox-maps-android` changelog: `<changelog></changelog>`.
 - [ ] If this PR is a `v10.[version]` release branch fix / enhancement, merge it to `main` firstly and then port to `v10.[version]` release branch.

### Summary of changes

Patch release v10.0.1 holding billing fix.

### CODE REVIEW

**Review by Chaoba — APPROVED** [review]
We also need cherry-pick the changelog to the main branch, right?

**Review by yunikkk — APPROVED** [review]

**Comment by kiryldz** [comment]
> We also need cherry-pick the changelog to the main branch, right?

Good point, yes, will do it afterwards.

**Review by alexander-kulikovskii — APPROVED** [review]

File: CHANGELOG.md [hunk 1]
```diff
@@ -2,6 +2,14 @@
 
 Mapbox welcomes participation and contributions from everyone.
 
+# 10.0.1 November 26, 2021
+
+## Bug fixes 🐞
+* Fix billing issue when upgrading Mapbox Maps SDK from v9 to v10.
```
Comment (ank27) [inline CHANGELOG.md:8]: was the issue only when you upgrade from hydrogen to carbon, or was it for older releases (eg v6 or something else) to carbon, just curious?

**Review by ank27 — APPROVED** [review]

**Review by pengdev — APPROVED** [review]

File: CHANGELOG.md (hunk 1 above)
Comment (pengdev) [reply CHANGELOG.md:8]: nit: add link to the PR as well?

Comment (kiryldz) [reply CHANGELOG.md:8]: I am not adding links here as there's one PR for it all 😄 
I'll add it though to release notes.

**Pull request merged**